components/raftstore-v2/src/raft/peer.rs

```diff
@@ -167,6 +187,57 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
         self.region().get_id()
     }
 
+    /// Set the region of a peer.
+    ///
+    /// This will update the region of the peer, caller must ensure the region
+    /// has been preserved in a durable device.
+    pub fn set_region(
+        &mut self,
+        // host: &CoprocessorHost<impl KvEngine>,
+        reader: &mut ReadDelegate,
+        region: metapb::Region,
+        reason: RegionChangeReason,
+        tablet_index: u64,
+    ) {
+        if self.region().get_region_epoch().get_version() < region.get_region_epoch().get_version()
+        {
+            // Epoch version changed, disable read on the local reader for this region.
+            //
+            // The difference of `expire` and `expire_remote_lease` is that `expire` expires
+            // lease both in local reader and raftstore while
+            // `expire_remote_lease` only expires the lease in local reader.
+            //
+            // V1 calls `expire_remote_lease` while here v2 calls `expire`. This
+            // difference is due to v2 only performs read by local reader and
+            // raftstore is just a place to renew the lease. If the lease in raftstore is
+            // not expired, it may not renew lease.
+            self.leader_lease.expire();
```

Comment:
No, v2 should be the same as v1. `expire` is just used to forbid on-going read. All local reader should use the new lease as the epoch change.